splinepy/microstructure/tiles/cross_3d.py

```diff
@@ -66,305 +70,407 @@ def _closing_tile(
                 * 0.2
             )
 
-        if parameter_sensitivities is not None:
-            raise NotImplementedError(
-                "Derivatives are not implemented for this tile yet"
-            )
-
         if not (_np.all(parameters > 0) and _np.all(parameters < 0.5)):
             raise ValueError("Thickness out of range (0, .5)")
 
+        # Check if user requests derivative splines
+        if parameter_sensitivities is not None:
+            self.check_param_derivatives(parameter_sensitivities)
+            n_derivatives = parameter_sensitivities.shape[2]
+            derivatives = []
+        else:
+            n_derivatives = 0
+            derivatives = None
+
         if not (0.0 < float(boundary_width) < 0.5):
             raise ValueError("Boundary Width is out of range")
 
         if not (0.0 < float(filling_height) < 1.0):
             raise ValueError("Filling must  be in (0,1)")
 
-        inv_boundary_width = 1.0 - boundary_width
-        inv_filling_height = 1.0 - filling_height
-        center_width = 1.0 - 2 * boundary_width
-        ctps_mid_height_top = (1 + filling_height) * 0.5
-        ctps_mid_height_bottom = 1.0 - ctps_mid_height_top
-        r_center = center_width * 0.5
-
-        spline_list = []
-        if closure == "z_min":
-            # The branch is located at zmin of current tile
-            branch_thickness = parameters[5, 0]
-            ctps_corner = _np.array(
-                [
-                    [0.0, 0.0, 0.0],
-                    [boundary_width, 0.0, 0.0],
-                    [0.0, boundary_width, 0.0],
-                    [boundary_width, boundary_width, 0.0],
-                    [0.0, 0.0, filling_height],
-                    [boundary_width, 0.0, filling_height],
-                    [0.0, boundary_width, filling_height],
-                    [boundary_width, boundary_width, filling_height],
-                ]
-            )
+        splines = []
+        for i_derivative in range(n_derivatives + 1):
+            if i_derivative == 0:
+                bound_width = boundary_width
+                fill_height = filling_height
+                inv_boundary_width = 1.0 - boundary_width
+                inv_filling_height = 1.0 - filling_height
+                center_width = 1.0 - (2 * boundary_width)
+                ctps_mid_height_top = (1.0 + filling_height) * 0.5
+                ctps_mid_height_bottom = 1.0 - ctps_mid_height_top
+                r_center = center_width * 0.5
+                v_zero = 0.0
```

Comment:
is `v_zero` equivalent of `kZero`?

Reply:
No, because `v_one` is sometimes also `kZero`

Reply:
`v_`ery interesting

Reply:
This is required to reset values in case derivatives of constants are computed